src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -919,9 +921,18 @@ MarkObjectsDistributedCreateCommand(List *addresses,
 		int forceDelegation = list_nth_int(forceDelegations, currentObjectCounter);
 		List *names = NIL;
 		List *args = NIL;
+		char *objectType = NULL;
 
-		char *objectType = getObjectTypeDescription(address, false);
-		getObjectIdentityParts(address, &names, &args, false);
+		if (IsManagementCommand)
+		{
+			names = namesArg;
```

Comment:
We need role name to mark a role distributed in workers. In our case we try to do this before the role is created for this transaction, so getObjectIdentityParts cannot find the role name. Because of that, I pass the name to this function.
I should probably do this in some better way tho.